pyhms/demes/single_pop_eas/common.py

```diff
@@ -0,0 +1,34 @@
+from typing import Protocol
+
+import numpy as np
+
+from ...core.population import Population
+
+
+class VariationalOperator(Protocol):
+    def __call__(self, population: Population) -> Population:
+        pass
```

Comment:
This should be an abstract class

Reply:
Why?

Reply:
From what I understood, it does not provide an implementation of its methods and is here to be inherited from by other classes.

Reply:
Yes, exactly. That's why I'm using Protocol. It acts as an interface: https://jellis18.github.io/post/2022-01-11-abc-vs-protocol/

Reply:
Ok, nice

Reply:
Ok, nice. I was not aware of this feature